### ISSUE TITLE
attempt to write a readonly database (code 1032 SQLITE_READONLY_DBMOVED[1032])

### ISSUE THREAD

**Issue by kevinranks** [react-native-async-storage/async-storage#422]
I have users receiving this error: 

`attempt to write a readonly database (code 1032 SQLITE_READONLY_DBMOVED[1032])`

Some Googling and #178 even mentions that this could be related to the default max storage size (6MB). However, I've already implemented this (we're on `v1.7.0` prior to `AsyncStorage_db_size_in_MB` implementation) into our `MainApplication.java` but it has had no affect:

```
@Override
public void onCreate() {
	super.onCreate();
	SoLoader.init(this, /* native exopackage */ false);
	long size = 50L * 1024L * 1024L; // 50 MB
	com.facebook.react.modules.storage.ReactDatabaseSupplier.getInstance(getApplicationContext()).setMaximumSize(size);
}
```

I verified that the size increase was working by implementing a custom NativeModule which calls `ReactDatabaseSupplier` then `getMaximumSize()` to retrieve the size and send it across the bridge to JavaScript. When inside my app and calling `getMaxSize()` it successfully returns 50 MB:

```
@ReactMethod
public void getMaxSize(Callback successCallback) {
	Long maxDatabaseSize = com.facebook.react.modules.storage.ReactDatabaseSupplier.getInstance(reactContext).get().getMaximumSize();
	successCallback.invoke(maxDatabaseSize.doubleValue());
}
```

Does this look correct for verifying that size? Is this error unrelated to Android AsyncStorage size? Looking through the Sentry breadcrumbs this error always seems to occur after the user backgrounds & foregrounds the app following by low memory:

![image](https://user-images.githubusercontent.com/6843831/90810127-b924c300-e2df-11ea-9925-bf8000b3b99b.png)

**Comment by Krizzu** [comment]
Hey @kevinranks 

I'm trying to address this in #423. I'm not able to reproduce this issue myself, so small research with similar problems points it to locale change/db connection not closed properly. Any help here is appreciated 🙏

**Comment by kevinranks** [comment]
@Krizzu We're also unable to reproduce the error despite a lot of trying. If your PR gets merged in we will definitely try it.

**Comment by baltagih2** [comment]
Same issue here

Error
attempt to write a readonly database (code 1032 SQLITE_READONLY_DBMOVED[1032])

```
node_modules/@react-native-community/async-storage/lib/AsyncStorage.js in convertError at line 413:19
function convertError(error): ?Error {
  if (!error) {
    return null;
  }
  const out = new Error(error.message);
  // $FlowFixMe: adding custom properties to error.
  out.key = error.key;
  return out;
}
```

"@react-native-community/async-storage": "1.8.0",

**Comment by tido64** [comment]
Addressed by #423

**Comment by hardikweetech** [comment]
> Hey @kevinranks
> 
> I'm trying to address this in #423. I'm not able to reproduce this issue myself, so small research with similar problems points it to locale change/db connection not closed properly. Any help here is appreciated 🙏

delete database and add copy of it on it's place manually
